### ISSUE TITLE
method_exists inconsistent with parameterised functions

### ISSUE THREAD

**Issue by mauro3** [JuliaLang/julia#8959]
I found the following inconsistence in `method_exists`:

``` julia
# without type parameters, all good:
f(x,y) = 1
@assert method_exists(f, (None, None))
@assert method_exists(f, (Int, None))
@assert method_exists(f, (Int, Int))
@assert method_exists(f, (None, Int))

# with two type parameters, all good:
h{T,K}(x::T,y::K) = 1
@assert method_exists(h, (None, None))
@assert method_exists(h, (Int, None))
@assert method_exists(h, (Int, Int))
@assert method_exists(h, (None, Int))

# with one type parameter:
g{T}(x::T,y) = 1
@assert method_exists(g, (None, None))
# ERROR: assertion failed: method_exists(g,(None,None))
#  in error at error.jl:21
@assert method_exists(g, (Int, None))
# ERROR: assertion failed: method_exists(g,(Int,None))
#  in error at error.jl:21
@assert method_exists(g, (Int, Int))
@assert method_exists(g, (None, Int))
```

I think the last example should work like the second one.

**Comment by mauro3** [comment]
Some more datapoints:
```
julia> @assert method_exists(f, (Any,Any))

julia> @assert method_exists(h, (Any,Any))
ERROR: AssertionError: method_exists(h,(Any,Any))

julia> @assert method_exists(g, (Any,Any))
ERROR: AssertionError: method_exists(g,(Any,Any))
```

**Comment by mauro3** [comment]
This was fixed in 0.4.  (The two AssertionErrors in my second post are correct and still present)
